### ISSUE TITLE
`useForm` needs generics

### ISSUE THREAD

**Issue by codyarose** [SelectQuoteLabs/SQForm#337]
The current types for `useForm` are not flexible enough for the way some components utilize it, and this is becoming a blocking issue for getting those components migrated to TS. We likely need type arguments for the field's value and event handlers at minimum.

**Comment by 20BBrown14** [comment]
Alright. It's been a hot minute since we've made any progress on this. But I _think_ I might have landed on something that is useable. The magic around generics + react events is still a little unknown to me but TS doesn't seem to complain and the types seem to line up with what I think they should be.

I have this branch linked below with the changes. I cribbed some of the changes from Cody's draft PR for this same issue.

After make the changes to the `useForm` hook I converted SQFormMultiSelect and SQFormDropdown to TS to display the changes working as expect. In the same vein, I updated some of the other components that benefit from using the generics.

Hopefully this will get some traction again and we can move forward with complete the Typescript conversion. I think this was the main blocker.

https://github.com/SelectQuoteLabs/SQForm/tree/issue/337_useForm_generics_revisited
